tests/test_cuda_targeted_insertion_mover.py

```diff
@@ -624,36 +744,46 @@ def test_tibd_exchange_deterministic_moves(radius, proposals_per_move, precision
         seed,
         proposals_per_move,
         1,
+        batch_size=batch_size,
     )
 
     iterative_moved_coords = conf.copy()
     for _ in range(proposals_per_move):
         iterative_moved_coords, _ = bdem_a.move(iterative_moved_coords, box)
         assert not np.all(conf == iterative_moved_coords)  # We should move every time since its a single mol
     batch_moved_coords, _ = bdem_b.move(conf, box)
-    # Moves should be deterministic regardless the number of steps taken per move
-    np.testing.assert_array_equal(iterative_moved_coords, batch_moved_coords)
-
     assert bdem_a.n_accepted() > 0
     assert bdem_a.n_proposed() == proposals_per_move
     assert bdem_a.n_accepted() == bdem_b.n_accepted()
     assert bdem_a.n_proposed() == bdem_b.n_proposed()
 
+    # Moves should be deterministic regardless the number of steps taken per move
+    np.testing.assert_array_equal(iterative_moved_coords, batch_moved_coords)
+    if batch_size == 1:
+        # Where the batch size is 1 the last log probabilities should match
+        if precision == np.float64:
+            # For some reason these disagree very slightly in float64,  something to dig into
```

Comment:
Yes, sorry I had meant to point this failure out more prominently when I opened the PR. 

`raw_log_probability_host` also does not match bitwise.

Reply:
Following up from offline discussion: There is a difference between the initial weights from `compute_initial_weights` and the weights that come from `compute_incremental_weights`. Going to make this identical so that there is no uncertainty.

Reply:
Following up on this, it appears to be caused by the fact that

```
_int128 foobar = static_cast<__int128>(39598530494464); // Value selected from seen in practice
foobar != FLOAT_TO_FIXED_ENERGY<RealType>(FIXED_ENERGY_TO_FLOAT<RealType>(foobar)
```

Which means that after we compute the log weights (fixed -> float) we are no longer able to maintain the same result with `compute_incremental_weights` as we do with `compute_initial_weights`. 

Think the right approach is to store all of the weights in __int128, will give that a try.

Reply:
The energies are all computed in fixed point now and converting from energies -> log weights is done only when we need the log weights